lib/paneron/register/raw/register.rb

```diff
@@ -366,7 +366,7 @@ def self.validate_path(path)
         def to_lutaml
           Paneron::Register::Register.new(
             data_sets: data_set_lutamls,
-            metadata: metadata.to_json,
+            metadata: Paneron::Register::RegisterMetadata.from_yaml(metadata.to_yaml),
```

Comment:
This is very weird... can we get rid of this?

Reply:
I agree.  Was just trying to fix something broken.  Lemme see if I can revert it.

Reply:
Looks not.  If I just put `metadata: metadata`, it'll raise this error:

```

  1) Paneron::Register::Register#metadata is expected to be a kind of Lutaml::Model::Serializable
     Failure/Error: it { is_expected.to be_kind_of(Lutaml::Model::Serializable) }
       expected {"datasets"=>{"reg-1"=>true, "reg-2"=>true, "reg-3"=>true}, "title"=>"register"} to be a kind of Lutaml::Model::Serializable
```

Reply:
My expectation was that LutaML model would do a deep serialization for me...  Maybe I'm missing something?

Reply:
@ribose-jeffreylau of course lutaml-model serializes the entire model hierarchy in full. Originally you had `to_json` which means `metadata` becomes a string? You can directly pass in the model?

Reply:
Do you mean `Paneron::Register::RegisterMetadata.new(metadata)`?

Reply:
Sorry I'm not well read into what exactly you're trying to do here... But if you want to pass a lutaml-model into an attribute of a lutaml-model, it should work properly, given that the parent model has the proper attribute type defined.

It seems that the code `YAML.load(register.metadata.to_yaml)` is giving you a `Hash`, hence lutaml-model is complaining you're giving it the wrong type of value.

Instead you should just pass on `register.metadata`.

Reply:
I think you were assuming that the constructor of a lutaml-model object will take "params" or "deep params". No, it doesn't. It takes the value of whatever you have defined as the type.

Instead, if you really need a `from_hash` (I'm not sure if we have it) to create the object of the right type, then it can be assigned as an attribute.